lucene/misc/src/java/org/apache/lucene/misc/store/DirectIODirectory.java

```diff
@@ -346,17 +360,21 @@ public byte readByte() throws IOException {
 
     private void refill() throws IOException {
       filePos += buffer.capacity();
-      
+
+      // BaseDirectoryTestCase#testSeekPastEOF test for consecutive read past EOF,
+      // hence throwing EOFException early to maintain buffer state (position in particular)
+      if(filePos >= channel.size()) {
```

Comment:
Thanks, this was driving me crazy when I tried to fix this (I gave up after some beers deep night).

Reply:
I think this was exactly the problem. I was yesterday about to change this to `>` instead of `>=`. It is allowed to seek to exectly the file size, which was obviously causing the test failures on windows.